### PR TITLE
Replace k8s.io/utils/pointer  with k8s.io/utils/ptr in e2e tests

### LINKED ISSUE #121690: Cleanup for apps e2e tests: do not use deprecated pointer package
/kind cleanup 
/good-first-issue
/sig-apps

Replace usages of "k8s.io/utils/pointer" with "k8s.io/utils/ptr".
The old one is currently used for:
- job: https://github.com/kubernetes/kubernetes/blob/227d1b2357d93a6884addccb50122df16674ca95/test/e2e/apps/job.go#L52
- stateful set: https://github.com/kubernetes/kubernetes/blob/227d1b2357d93a6884addccb50122df16674ca95/test/e2e/apps/statefulset.go#L60
- controller_revision: https://github.com/kubernetes/kubernetes/blob/227d1b2357d93a6884addccb50122df16674ca95/test/e2e/apps/controller_revision.go#L44

### PR DESCRIPTION
**Pull request by Sakuralbj**
#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #121690

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/apps/job.go [hunk 1]
```diff
@@ -276,7 +275,7 @@ var _ = SIGDescribe("Job", func() {
 	ginkgo.It("should not create pods when created in suspend state", func(ctx context.Context) {
 		ginkgo.By("Creating a job with suspend=true")
 		job := e2ejob.NewTestJob("succeed", "suspend-true-to-false", v1.RestartPolicyNever, parallelism, completions, nil, backoffLimit)
-		job.Spec.Suspend = pointer.BoolPtr(true)
+		job.Spec.Suspend = ptr.To[bool](true)
```
Comment (mimowo) [inline test/e2e/apps/job.go:278]: I think we can drop "bool". If so, drop it in other places as well.

**Comment by kannon92** [comment]
/ok-to-test

Once you fix the bool comment from @mimowo I think this is good to go.

File: test/e2e/apps/job.go (hunk 1 above)
Comment (Sakuralbj) [reply test/e2e/apps/job.go]: OK. I fix it

**Review by mimowo — COMMENTED** [review]
/lgtm
/assign @soltysh

**Comment by mimowo** [comment]
cc @pohly

**Comment by mimowo** [comment]
@Sakuralbj thanks for this work. 

However, we may need to close the PR due to the strategy for handling this depreciation. It is currently being discussed here: https://app.slack.com/client/T09NY5SBT/CHGFYJVAN/1699022254.366619. FYI @pohly.

**Comment by Sakuralbj** [comment]
> @Sakuralbj thanks for this work.
> 
> However, we may need to close the PR due to the strategy for handling this depreciation. It is currently being discussed here: https://app.slack.com/client/T09NY5SBT/CHGFYJVAN/1699022254.366619. FYI @pohly.

Ok

**Comment by mimowo** [comment]
/close
As per slack thread. The old packages are meant to stay, so there is no need to switch. If it makes coding easier, the switch can be justified as a preparatory PR.

**Pull request closed without merging**